pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -302,7 +302,7 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusters(cb *ClusterBuilder, i
 		for _, instances := range clustersToBuild {
 			instance := instances[0]
 			localCluster := cb.buildInboundClusterForPortOrUDS(cb.proxy, int(instance.Endpoint.EndpointPort), actualLocalHost, instance, instances)
-			clusters = cp.conditionallyAppend(clusters, localCluster)
+			clusters = cp.conditionallyAppend(clusters, instance.Service.Hostname, localCluster)
```

Comment:
This doesn't work quite right I think. There can be multiple hostnames associated with a single inbound cluster. That was the whole motivation for the cluster name change

Reply:
hmm.. Should we try across all instances?

Reply:
@howardjohn The latest commit should fix it. PTAL.